src/dune_rules/lock_dir.ml

```diff
@@ -25,17 +25,40 @@ module Load = Make_load (struct
     ;;
   end)
 
+let select_lock_dir lock_dir_selection =
+  let* workspace = Workspace.workspace () in
+  let* raw_expander =
+    Memo.of_non_reproducible_fiber
+      (Sys_poll.make ~path:(Env_path.path Env.initial)
```

Comment:
You can share this computation with what we have in `pkg_rules` I think. That would make it so that we don't evaluate these variables more than once per run.

Reply:
Moved the logic into lock_dir.ml and reused in pkg_rules.ml